tracing-appender/src/worker.rs

```diff
@@ -74,16 +74,13 @@ impl<T: Write + Send + Sync + 'static> Worker<T> {
                     Ok(WorkerState::Continue) | Ok(WorkerState::Empty) => {}
                     Ok(WorkerState::Shutdown) | Ok(WorkerState::Disconnected) => {
                         let _ = self.shutdown.recv();
-                        break;
+                        return;
                     }
                     Err(_) => {
                         // TODO: Expose a metric for IO Errors, or print to stderr
                     }
                 }
             }
-            if let Err(e) = self.writer.flush() {
-                eprintln!("Failed to flush. Error: {}", e);
-            }
```

Comment:
i'm assuming the flush here was removed because we will already always flush on shutdown in `Worker::work()`?

Reply:
that's why I felt safe to remove it

but the actual reason was that it could block (or introduce undefined delay)(https://github.com/tokio-rs/tracing/pull/1125#issuecomment-742873844) when joining the worker. To be honest, I would've just joined the thread, but the IO argument does make sense and I just followed with it.

Reply:
I realised that within this logic it might be a good idea to drop writer as well: 45da5122a92b8820f674c3962ff67605d40b510f. There could be some IO there in destructor.